### PR TITLE
MTN use reusable workflow and replace deprecated to_dict

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**

### CODE REVIEW

File: solvers/chop.py [hunk 1]
```diff
@@ -2,6 +2,10 @@
 from benchopt import BaseSolver, safe_import_context
 
 with safe_import_context() as import_ctx:
+    # see https://github.com/pytorch/pytorch/issues/78490
+    import os
+    os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'
```
Comment (Badr-MOUFAD) [inline solvers/chop.py:7]: A fix suggested in https://github.com/benchopt/benchmark_logreg_l2/pull/37

**Comment by Badr-MOUFAD** [comment]
I believe these changes fix the CI.
Any idea @tomMoral why the test takes so much time on macOS? they lasted for about 12 in Linux.

**Review by tomMoral — COMMENTED** [review]
The CI has been fixed by merging #37, thanks for pointing out that the change was missing @Badr-MOUFAD 
I think for this Pr, it is valuable to move to the shared workflow and `get_objective` but maybe revert the changes in solvers which seems unrelated?

File: solvers/cuml.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ class Solver(BaseSolver):
     requirements = [
         "rapidsai::rapids",
         f"nvidia::cudatoolkit={cuda_version}",
-        "dask-sql",
+        "conda-forge::dask-sql",
```
[the remark this answers is not in the document]
Comment (Badr-MOUFAD) [reply solvers/cuml.py:24]: My bad

File: solvers/lightning.py [hunk 3]
```diff
@@ -10,7 +10,8 @@ class Solver(BaseSolver):
 
     install_cmd = 'conda'
     requirements = [
-        'pip:git+https://github.com/scikit-learn-contrib/lightning.git'
+        'pip:Cython',
+        'pip:git+https://github.com/scikit-learn-contrib/lightning.git@master'
```
[the remark this answers is not in the document]
Comment (Badr-MOUFAD) [reply solvers/lightning.py:14]: I remember getting an error when installing Lightning as it needs to compile/build Cython ext

**Comment by Badr-MOUFAD** [comment]
Despite the changes, the CI tests on macOS machines take a long time [to be canceled in the end](https://github.com/benchopt/benchmark_logreg_l2/actions/runs/5028212896/jobs/9018654476?pr=38).

I can't reproduce locally as I m on Linux.
@tomMoral is it possible to put a timeout on the solvers' test to prevent them from everlasting?

**Comment by tomMoral** [comment]
The build is canceled after 5h waiting for the last test to start.
image

File: test_config.py [hunk 4]
```diff
@@ -12,8 +12,13 @@ def check_test_solver_install(solver_class):
...
+
     if solver_class.name.lower() == "cuml":
-        if sys.platform == "darwin":
+        if is_platform_macOS:
             pytest.xfail("Cuml is not supported on MacOS.")
         if cuda_version is None:
             pytest.xfail("Cuml needs a working GPU hardware.")
+
+    if is_platform_macOS and "snapml" in solver_class.name.lower():
+        pytest.xfail("running snapml on MacOS takes a lot of time.")
```
Comment (tomMoral) [inline test_config.py:24]: can you skip only the `test_solver` test?
(add a function `check_test_solver(solver_class)` in this file I believe)

**Comment by tomMoral** [comment]
hum, can you check the test is indeed skipped? It seems that the first solution (skipping the install) worked but this one still deadlock...

**Comment by tomMoral** [comment]
It was indeed stuck, seems to have issues in closing the python interpreter...
Let's go with your original solution @Badr-MOUFAD , sorry for the bike shedding.

**Review by tomMoral — APPROVED** [review]
LGTM

**Pull request merged**